src/modules/piano.py

```diff
@@ -31,7 +31,7 @@ def draw(self, img):
         return img
 
     def key_generator(self, spath, octave, key_num):
-        if (key_num > 14) or (octave != 4):
+        if (key_num > 14) or (octave not in [3, 4]):

```

Comment:
You have 3 octaves - 3, 4, 5
In that case, you need to do:
* `key_num > 21`
* `octave not in [3, 4, 5]`

Reply:
if we use 21 key_num and octave 4 or 5, we go beyond the border

Reply:
so either we write a very large condition on the keys, or we leave some restriction at this stage

Reply:
why? I have just tried this and it works. Try it on your machine

Reply:
this draws but crashes when trying to press a key

Reply:
because we don't have a sixth octave